providers/openfeature-provider-ofrep/src/openfeature/contrib/provider/ofrep/__init__.py

```diff
@@ -124,7 +130,7 @@ def _resolve(
 
         try:
             response = self.session.post(
-                urljoin(self.base_url, f"/ofrep/v1/evaluate/flags/{flag_key}"),
+                urljoin(self._get_ofrep_api_url(), f"evaluate/flags/{flag_key}"),
```

Comment:
Based on this stackoverflow: https://stackoverflow.com/a/1793282/3708208, urljoin interprets a string starting with a slash as a path on the server root. Maybe we don't need the method at all, but we need to change the second parameter of the urljoin.

```suggestion
                urljoin(self.base_url, f"ofrep/v1/evaluate/flags/{flag_key}"),
```

Reply:
I had thought about this as well. The only issue is that, if a user provides a base_url which does not end in a trailing slash then the last part of their provided base_url will get dropped by `urljoin()`. For example, `urljoin("https://localhost:8080/tools/featureFlags", "ofrep/v1/keys")` becomes `'https://localhost:8080/tools/ofrep/v1/keys'`. 

We can add in some logic to ensure that the base_url ends with a slash and have it all in `_resolve`. I just broke it out into a separate function so that it was easier to create the unit tests for validating that the ofrep provider's url parsing behaviour remains consistent and doesn't get broken over time

Reply:
Ah understood!